### ISSUE TITLE
[Regression] Godot doesn't open anymore

### ISSUE THREAD

**Issue by guilhermefelipecgs** [godotengine/godot#15068]
**Godot version:**
4a2eef4ad801a3dc5c07a575e59b8f1eafa42c6e

**OS/device including version:**
Linux 4.14.8, Window Manager: i3

**Issue description:**
I can't open godot after this commit 6f3486c4866a0a0e0408c2e0dbd65f128995abf4 on i3wm.
On i3, when a I open the project, I get stuck in this loop.
https://github.com/godotengine/godot/blob/master/platform/x11/os_x11.cpp#L1053

**Minimal reproduction project:**

- [x] I searched the existing [GitHub issues](https://github.com/godotengine/godot/issues?utf8=%E2%9C%93&q=is%3Aissue+) for potential duplicates.

**Comment by reduz** [comment]
This bug is impossible to fix, every time we try to fix the maximized splash problem we run into strange behaviors :(

**Comment by Timofffee** [comment]
Similarly. Also i3wm
Output:
```
  X Error of failed request:  BadWindow (invalid Window parameter)
  Major opcode of failed request:  20 (X_GetProperty)
  Resource id in failed request:  0x3a00002
  Serial number of failed request:  1255558
  Current serial number in output stream:  1255558
```

**Comment by groud** [comment]
I already commented this saying this while loop is dangerous. A maximum number of iterations / seconds would have been added to the check. 
Sadly, @Hinsbart merged it before we could discuss it :p

**Comment by oisincar** [comment]
Just confirming commenting out that block/commit fixes the issue for me on xmonad too.

**Comment by rminderhoud** [comment]
Can we revert that commit?

**Comment by oisincar** [comment]
Added a PR for @groud's idea of a timeout for the loop.

**Comment by AlmightyScientist** [comment]
I will fall in love with X11. Kappa.

https://github.com/godotengine/godot/blob/c899583ac1bd9c1c424ec2536988ad0790526eea/platform/x11/os_x11.cpp#L1715-L1723

You don't say, @reduz

**Comment by AlmightyScientist** [comment]
Arf, I knew this while loop wasn't safe, but didn't except it would be this troublesome..

**Comment by lehthanis** [comment]
Also affected. I3wm. Pressing the mod-f fullscreen command doesn't make it exit loop. Has anyone tried setting godot to auto maximize in i3? Worth a shot?

**Comment by guilhermefelipecgs** [comment]
@lehthanis Technically, i3wm (and maybe all tiling wm) doesn't support maximize, even if you go to fullscreen. I already did a fix #15092 for this.
